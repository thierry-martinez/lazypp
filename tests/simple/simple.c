#define A 1
( A )
#define B(x) {x}
B(bla)

B (bla)

#ifdef A
OK
#else
BOOM
#endif

#ifdef unknown /* comment */
maybe
#endif

#if defined(A)
OK
#endif

#define __GLIBC_USE(X) __GLIBC_USE_ ## X

#if __GLIBC_USE (IEC_60559_BFP_EXT) || __GLIBC_USE (ISOC2X)
OK
#endif

#define answer 42

#if answer < 10
  A
#else
  B
#endif

#if defined __USE_GNU || defined __STDC_WANT_IEC_60559_FUNCS_EXT__
  A
#endif

#define X A

#define Y(A) B(A)
