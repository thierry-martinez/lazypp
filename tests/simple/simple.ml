let check_file ~source_filename ~target_filename =
  let expected_filename = source_filename ^ ".expected" in
  let diff_result =
    Sys.command
      (Filename.quote_command "diff"
         [target_filename; expected_filename]) in
  if diff_result = 0 then
    Ok ()
  else
    Error ()

let () =
  let config : Lazypp.Config.t = {
      paths = [];
      defines = [];
      undefines = [];
      default_undefined = false;
      predefs = [];
      use_gcc_predefs = false;
    } in
  let handlers = Lazypp.Output.default_handlers in
  match LazyppTest.test_all_c_files config handlers check_file with
  | Ok () -> ()
  | Error () -> exit 1
