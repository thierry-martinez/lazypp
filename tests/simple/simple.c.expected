#define A 1
( 1 )
#define B(x) {x}
{bla}

{bla}

OK

#ifdef unknown /* comment */
maybe
#endif

OK

#define __GLIBC_USE(X) __GLIBC_USE_ ## X

#if __GLIBC_USE_IEC_60559_BFP_EXT || __GLIBC_USE_ISOC2X
OK
#endif

#define answer 42

  B

#if defined __USE_GNU || defined __STDC_WANT_IEC_60559_FUNCS_EXT__
  1
#endif

#define X 1

#define Y(A) {A}
